oak_crypto/src/util.rs

```diff
@@ -0,0 +1,59 @@
+//
+// Copyright 2023 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use alloc::{vec, vec::Vec};
+use anyhow::anyhow;
+use core::cmp::Ordering;
+
+pub(crate) fn xor<const L: usize>(first: &[u8; L], second: &[u8; L]) -> anyhow::Result<[u8; L]> {
+    first
+        .iter()
+        .zip(second.iter())
+        .map(|(&x, &y)| x ^ y)
+        .collect::<Vec<u8>>()
+        .try_into()
+        .map_err(|error| anyhow!("couldn't run xor operation: {:?}", error))
```

Comment:
Maybe it's worth doing it in a non-functional way, it's kind of odd to expect that an xor may fail, and here it's just because the type system loses track of the lengths, but we statically know that they are all the same. Also there is an unnecessary Vec allocation.

Basically just allocate the return slice and fill it in byte by byte.

Reply:
Done